scripts/uberenv/packages/hdf5/package.py

```diff
@@ -0,0 +1,381 @@
+            # two paths: the first one points to the directory with include
+            # files, the second one points to the directory with library files.
+            # If the second path is not specified, the configure script assumes
+            # that it equals to prefix/lib. However, the correct directory
+            # might be prefix/lib64. It is not a problem when the building is
+            # done with Spack's compiler wrapper but it makes the Libtool
+            # files (*.la) invalid, which makes it problematic to use the
+            # installed library outside of Spack environment.
+            extra_args.append('--with-szlib=%s,%s' %
+                              (szip_spec.headers.directories[0],
+                               szip_spec.libs.directories[0]))
+        else:
+            extra_args.append('--without-szlib')
+
+        if self.spec.satisfies('@1.10:'):
+            if '+debug' in self.spec:
+                extra_args.append('--enable-build-mode=debug')
+            else:
+                extra_args.append('--enable-build-mode=production')
+        else:
+            if '+debug' in self.spec:
+                extra_args.append('--enable-debug=all')
+            else:
+                extra_args.append('--enable-production')
+
+            # '--enable-fortran2003' no longer exists as of version 1.10.0
+            if '+fortran' in self.spec:
+                extra_args.append('--enable-fortran2003')
+            else:
+                extra_args.append('--disable-fortran2003')
+
+        if '+shared' in self.spec:
+            extra_args.append('--enable-shared')
+        else:
+            extra_args.append('--disable-shared')
+            extra_args.append('--enable-static-exec')
+
+        if '+pic' in self.spec:
+            extra_args.extend([
+                'CFLAGS='   + self.compiler.cc_pic_flag,
+                'CXXFLAGS=' + self.compiler.cxx_pic_flag,
+                'FCFLAGS='  + self.compiler.fc_pic_flag,
+            ])
+
+        if '+mpi' in self.spec:
+            # The HDF5 configure script warns if cxx and mpi are enabled
+            # together. There doesn't seem to be a real reason for this, except
+            # that parts of the MPI interface are not accessible via the C++
+            # interface. Since they are still accessible via the C interface,
+            # this is not actually a problem.
+            extra_args += ['--enable-parallel',
+                           'CC=%s' % self.spec['mpi'].mpicc]
+
+            if '+cxx' in self.spec:
+                extra_args.append('CXX=%s' % self.spec['mpi'].mpicxx)
+
+            if '+fortran' in self.spec:
+                extra_args.append('FC=%s' % self.spec['mpi'].mpifc)
+
+        extra_args.append('--with-zlib=%s' % self.spec['zlib'].prefix)
+ 
+        #SERAC EDIT BEGIN
+        #for some reason, it needs -pthread on WSL
+        extra_args.append('LDFLAGS=-pthread')
```

Comment:
@becker33 is there a better way of doing this? Adding -pthread to the global spack LDFLAGS caused problems.

Reply:
I realized I tried `ldflags` instead of `ldlibs`. The latter works as a compiler-specific fix.

Reply:
I wonder if its an ordering issue between ldflags and ldlibs, since according to the gnu docs they are both given to the compiler for the linker.  Anyways, thanks for the fix.